### ISSUE TITLE
Code coverage error should not fail a build

### ISSUE THREAD

**Issue by mikhailshilkov** [pulumi/pulumi#9730]
Failures like this one: https://github.com/pulumi/pulumi/runs/6614209710?check_suite_focus=true#step:40:25

![image](https://user-images.githubusercontent.com/1454008/170940387-0138eb65-e612-4c5a-9d84-5b0690860227.png)

should not fail the whole build.

**Comment by RobbieMcKinstry** [comment]
I had this same bug with a personal project just last week (though I didn't get to the bottom of it). I suspect "fail CI if error" is the [config toggle](https://github.com/codecov/codecov-action#arguments) we want, at the Action level.

**Comment by RobbieMcKinstry** [comment]
I submitted [an upstream PR](https://github.com/codecov/codecov-action/pull/769) to solve the issue we encountered.
TL;DR: GH Action input`fail_ci_on_error` wasn't respected in some instances.

I'm going to move this issue in the tracker to "blocked" since we're awaiting PR review.
